api/src/db/migrations/v127_create_remote_worker_seller_phase.test.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  generateV126Business,
+  generateV126ProfileData,
+  generateV126UserData,
+} from "@db/migrations/v126_remove_foreign_business_type";
+import { migrate_v126_to_v127 } from "@db/migrations/v127_create_remote_worker_seller_phase";
+
+describe("migrate_v126_to_v127", () => {
+  it("sets operatingPhase for Remote Sellers to 'REMOTE_SELLER_WORKER'", () => {
+    const id = "biz-1";
+    const v126ProfileData = generateV126ProfileData({});
+    const v126Business = generateV126Business({
+      id,
+      profileData: {
+        ...v126ProfileData,
+        operatingPhase: "NEEDS_TO_FORM",
+        foreignBusinessTypeIds: ["revenueInNJ"],
+      },
+    });
+    const v126UserData = generateV126UserData({ businesses: { "biz-1": v126Business } });
+
+    const v127 = migrate_v126_to_v127(v126UserData);
+    expect(v127.businesses[id].profileData.operatingPhase).toEqual("REMOTE_SELLER_WORKER");
+  });
+
+  it("sets operatingPhase for Remote Workers to 'REMOTE_SELLER_WORKER'", () => {
+    const id = "biz-1";
+    const v126ProfileData = generateV126ProfileData({});
+    const v126Business = generateV126Business({
+      id,
+      profileData: {
+        ...v126ProfileData,
+        operatingPhase: "NEEDS_TO_FORM",
+        foreignBusinessTypeIds: ["employeesInNJ"],
+      },
+    });
+    const v126UserData = generateV126UserData({ businesses: { "biz-1": v126Business } });
+
+    const v127 = migrate_v126_to_v127(v126UserData);
+    expect(v127.businesses[id].profileData.operatingPhase).toEqual("REMOTE_SELLER_WORKER");
+  });
+
+  it("does not modify operatingPhase for other Foreign businesses", () => {
+    const foreignBusinessTypeIds = [
+      "none",
+      "employeeOrContractorInNJ",
+      "officeInNJ",
+      "propertyInNJ",
+      "companyOperatedVehiclesInNJ",
+    ];
+    const randomIndex = Math.floor(Math.random() * foreignBusinessTypeIds.length);
```

Comment:
pebble - we've moved away from single test that randomly selects a value, reason being is that when a test fails and then passes when re-running, it's now clear why it's failing. We typically test for ever case.

Reply:
Great point